html/includes/graphs/multiport/auth.inc.php

```diff
@@ -1,10 +1,8 @@
 <?php
 
-$auth = true;
-
 foreach (explode(',', $vars['id']) as $ifid) {
-    if (!$auth && !port_permitted($ifid)) {
-        $auth = false;
+    if ($auth || port_permitted($ifid)) {
+        $auth = true;
```

Comment:
Would this not be worth writing as
if (port_permitted($ifid)) {
  $auth = true;
  break;
}